### PR TITLE
Publish on merge and tag

### PR DESCRIPTION
**Pull request by satabin**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -3,15 +3,18 @@ val commonSettings = Seq(
...
-      List(
-        compilerPlugin("org.typelevel" % "kind-projector" % "0.13.0" cross CrossVersion.full),
-        compilerPlugin("com.olegpy" % "better-monadic-for" % "0.3.1" cross CrossVersion.binary)
-      )
-  }
+  libraryDependencies ++= PartialFunction
+    .condOpt(CrossVersion.partialVersion(scalaVersion.value)) {
+      case Some((2, _)) =>
+        List(
+          compilerPlugin("org.typelevel" % "kind-projector"     % "0.13.0" cross CrossVersion.full),
```
Comment (ybasket) [inline build.sbt:10]:
```suggestion
          compilerPlugin("org.typelevel" % "kind-projector"     % "0.13.2" cross CrossVersion.full),
```

File: build.sbt [hunk 2]
```diff
@@ -22,11 +25,28 @@ val noPublish = List(
 
 // === CI/CD settings ===
 val scala213 = "2.13.6"
-val scala3 = "3.0.0"
+val scala3   = "3.0.0"
```
Comment (ybasket) [inline build.sbt:28]:
```suggestion
val scala3   = "3.0.2"
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -120,5 +121,9 @@ jobs:
           tar xf targets.tar
           rm targets.tar
 
-      - name: Publish project
-        run: sbt --client '++${{ matrix.scala }}; +publish'
\ No newline at end of file
+      - env:
```
Comment (ybasket) [inline .github/workflows/ci.yml:124]:
```suggestion
      - name: Publish project
        env:
```

**Review by ybasket — APPROVED** [review]
Left a few minor suggestions as I noticed some older versions, LGTM nevertheless

File: .github/workflows/ci.yml (hunk 3 above)
Comment (satabin) [reply .github/workflows/ci.yml:124]: This is generated file, I wouldn't edit it by hand :)

**Review by ybasket — APPROVED** [review]

**Pull request merged**